clouddriver-web/src/main/groovy/com/netflix/spinnaker/clouddriver/controllers/ArtifactController.java

```diff
@@ -48,4 +57,25 @@ public ArtifactController(Optional<ArtifactCredentialsRepository> artifactCreden
       return artifactCredentialsRepository.getAllCredentials();
     }
   }
+
+  @RequestMapping(method = RequestMethod.GET, value = "/fetch")
+  String fetch(@RequestParam("artifactAccount") String artifactAccount,
+      @RequestParam("type") String type,
+      @RequestParam("reference") String reference) {
+    if (artifactDownloader == null) {
+      throw new IllegalStateException("Artifacts have not been enabled. Enable them using 'artifacts.enabled' in clouddriver");
+    }
+
+    Artifact artifact = Artifact.builder()
+        .type(type)
+        .artifactAccount(artifactAccount)
+        .reference(reference)
+        .build();
+
+    try {
+      return IOUtils.toString(artifactDownloader.download(artifact));
```

Comment:
(just noticed this while cutting a release)

Thoughts on just sending the underlying stream all the way across to the caller without first converting to a `String`?

Similar to what was done here: https://github.com/spinnaker/clouddriver/blob/master/clouddriver-web/src/main/groovy/com/netflix/spinnaker/clouddriver/controllers/DataController.groovy#L55

Reply:
Oh sweet, didn't know that was possible - I'll switch that over.